resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java

```diff
@@ -211,7 +211,14 @@ private static Map<String, DnsServerAddresses> parse(File... etcResolverFiles) t
                                 port = Integer.parseInt(maybeIP.substring(i + 1));
                                 maybeIP = maybeIP.substring(0, i);
                             }
-                            addresses.add(SocketUtils.socketAddress(maybeIP, port));
+                            InetSocketAddress addr = SocketUtils.socketAddress(maybeIP, port);
+                            // Check if the address is resolved and only if this is the case use it. Otherwise just
+                            // ignore it. This is needed to filter out invalid entries, as if for example an ipv6
+                            // address is used with a scope that represent a network interface that does not exists
+                            // on the host.
+                            if (!addr.isUnresolved()) {
+                                addresses.add(SocketUtils.socketAddress(maybeIP, port));
```

Comment:
We can reuse the `addr` variable here, instead of recreating the `InetSocketAddress` object.